src/video/wayland/SDL_waylandwindow.c

```diff
@@ -898,7 +1080,24 @@ Wayland_SetWindowBordered(_THIS, SDL_Window * window, SDL_bool bordered)
 void
 Wayland_SetWindowResizable(_THIS, SDL_Window * window, SDL_bool resizable)
 {
-    CommitMinMaxDimensions(window);
+#ifdef HAVE_LIBDECOR_H
+    SDL_VideoData *data = _this->driverdata;
+    const SDL_WindowData *wind = window->driverdata;
+
+    if (data->shell.libdecor) {
+        if (wind->shell_surface.libdecor.frame == NULL) {
+            return; /* Can't do anything yet, wait for ShowWindow */
+        }
+        if (resizable) {
+            libdecor_frame_set_capabilities(wind->shell_surface.libdecor.frame, LIBDECOR_ACTION_RESIZE);
+        } else {
+            libdecor_frame_unset_capabilities(wind->shell_surface.libdecor.frame, LIBDECOR_ACTION_RESIZE);
+        }
+    } else
+#endif
+    {
+        CommitMinMaxDimensions(window);
```

Comment:
You still need to call `CommitMinMaxDimensions()` even in the libdecor case, no?

If minimum/maximum size changes were made while the window was not resizable, then min/max size as far as libdecor is concerned will currently be set to ` window->w` and `window->h`, rather than the appropriate `window->min_*` and `window->max_*` values.

Reply:
Agreed, if not the full function call we at least need to set window->max_w/h here, but it'd probably be better to call CommitMinMaxDimensions here for maximum safety.

Reply:
Removing the `LIBDECOR_ACTION_RESIZE` capability will already cause setting the min/max limits to the current content dimensions (as represented within libdecor). `CommitMinMaxDimensions` will also set the min/max limits, but those used by SDL. So I don't think we should set min/max limits twice.

Reply:
This part is just an unfortunate API redundancy in libdecor - if resizability was inferred from min/max we could just reuse the same code path, but from what it sounds like we're instead expected to let libdecor do this work for us. It's not great, but if that's a responsibility libdecor wants, so be it... this can be resolved.